core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -805,6 +813,16 @@ public static void exhaust(InputStream in) throws IOException {
     while (in.read(buf) != -1) {}
   }
 
+  /**
+   * Some status codes from the control plane are not appropritate to use in the data plane. If one
+   * is given it will be replaced with INTERNAL, indicating a bug in the control plane
+   * implementation.
+   */
+  public static Status replaceInappropriateControlPlaneStatus(Status status) {
+    checkArgument(status != null);
+    return INAPPROPRIATE_CONTROL_PLANE_STATUS.contains(status) ? Status.INTERNAL : status;
```

Comment:
Equality for status is reference equality, so `Status.OK.withDescription("Okay")` wouldn't match. See https://grpc.github.io/grpc-java/javadoc/io/grpc/Status.html#equals-java.lang.Object- (where it is explicitly not defined).

You'll want to add a test for that.